fearless/Modules/SoraCard/KYCMain/KYCMainInteractor.swift

```diff
@@ -85,54 +87,56 @@ private extension KYCMainInteractor {
 
     func getSoraChainAsset() {
         chainAssetFetching.fetch(
-            filters: [.assetName("XOR"), .chainId(Chain.soraMain.genesisHash)],
+            filters: [.assetName("XOR")],
             sortDescriptors: []
         ) { [weak self] result in
-            guard let result = result,
-                  case let .success(chainAssets) = result,
-                  let xorChainAsset = chainAssets.first
+            guard let strongSelf = self,
+                  let result = result,
+                  case let .success(chainAssets) = result
             else {
                 return
             }
-            self?.xorChainAsset = xorChainAsset
-            self?.getXorBalance(for: xorChainAsset)
-            self?.subscribeToPrice(for: xorChainAsset)
+            #if DEBUG
+                strongSelf.xorChainAssets = chainAssets
+            #else
+                strongSelf.xorChainAssets = chainAssets.filter { chainAsset in
+                    chainAsset.chain.chainId == Chain.soraMain.genesisHash
+                }
+            #endif
+            strongSelf.getXorBalance(for: strongSelf.xorChainAssets)
+            strongSelf.subscribeToPrice(for: strongSelf.xorChainAssets.first)
         }
     }
 
-    func getXorBalance(for chainAsset: ChainAsset) {
-        guard let accountId = wallet.fetch(for: chainAsset.chain.accountRequest())?.accountId
-        else {
-            return
-        }
-        accountInfoFetching.fetch(
-            for: chainAsset,
-            accountId: accountId
-        ) { [weak self] _, accountInfo in
-            self?.xorAccountInfo = accountInfo
+    func getXorBalance(for chainAssets: [ChainAsset]) {
+        accountInfoFetching.fetch(for: chainAssets, wallet: wallet) { [weak self] chainAssetsAccInfo in
```

Comment:
do you really need few balances there ? it looks like you need only xor balance. please check is it possible to fetch single balance with 'accountInfoFetching'